includes/handler/class-collection-sync.php

```diff
@@ -0,0 +1,182 @@
+<?php
+/**
+ * Collection Sync file.
+ *
+ * @package Activitypub
+ */
+
+namespace Activitypub\Handler;
+
+use Activitypub\Collection\Followers;
+use Activitypub\Collection\Remote_Actors;
+use Activitypub\Signature;
+
+use function Activitypub\get_url_authority;
+
+/**
+ * Collection Sync Handler.
+ *
+ * Handles the Collection-Synchronization header (FEP-8fcf) for various collection types.
+ *
+ * @see https://codeberg.org/fediverse/fep/src/branch/main/fep/8fcf/fep-8fcf.md
+ */
+class Collection_Sync {
+	/**
+	 * Initialize the class, registering WordPress hooks.
+	 */
+	public static function init() {
+		\add_action( 'activitypub_inbox_create', array( self::class, 'handle_collection_synchronization' ), 10, 2 );
+
+		\add_filter( 'http_request_args', array( self::class, 'maybe_add_headers' ), -1, 2 );
+	}
+
+	/**
+	 * Process Collection-Synchronization header if present (FEP-8fcf).
+	 *
+	 * This method handles the FEP-8fcf Collection Synchronization protocol for any collection type.
+	 * It detects the collection type from the URL and delegates to the appropriate handler.
+	 *
+	 * @see https://codeberg.org/fediverse/fep/src/branch/main/fep/8fcf/fep-8fcf.md
+	 *
+	 * @param array $data    The activity data.
+	 * @param int   $user_id The local user ID.
+	 */
+	public static function handle_collection_synchronization( $data, $user_id ) {
+		if ( empty( $_SERVER['HTTP_COLLECTION_SYNCHRONIZATION'] ) ) {
+			return;
+		}
+
+		// Check if sync-header is part of signature (required by FEP).
+		// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+		$signature = \wp_unslash( $_SERVER['HTTP_SIGNATURE'] ?? $_SERVER['HTTP_AUTHORIZATION'] ?? '' );
+		if ( false === \stripos( $signature, 'collection-synchronization' ) ) {
+			return;
+		}
+
+		// phpcs:ignore WordPress.Security.ValidatedSanitizedInput
+		$sync_header = \wp_unslash( $_SERVER['HTTP_COLLECTION_SYNCHRONIZATION'] );
+
+		// Parse the header using the generic HTTP parser.
+		$params = Signature::parse_collection_sync_header( $sync_header );
+
+		if ( false === $params ) {
+			return;
+		}
+
+		// Check for followers collection.
+		$collection_type = null;
+		if ( preg_match( '#/followers(?:/sync)?(?:\?|$)#', $params['url'] ) ) {
+			$collection_type = 'followers';
+		}
+
+		if ( ! $collection_type ) {
+			// Unknown or unsupported collection type.
+			return;
+		}
+
+		// Get the actor URL for validation.
+		$actor_url = isset( $data['actor'] ) ? $data['actor'] : null;
```

Comment:
```suggestion
		$actor_url = $data['actor'] ?? false;
```

Not sure if that's much better. Could also have the next conditional check `( ! isset( $data['actor'] ) )` and then use `$data['actor']` in subsequent calls.